### DISCUSSION TITLE
symfony-demo (or another demo) based on MicroSymfony

### DISCUSSION THREAD

**Discussion by tacman** [strangebuzz/MicroSymfony#5] [General]
This rocks.  I'm particular enthusiastic to see AssetMapper included, I think ux/live-components/assetmapper will make front-end development and deployment so much easier.

You mention Symfony Demo in the blog post, indeed, it has been the starting point for many developers to see best practices in action.  I'd love to see a version of the Symfony Demo for Symfony 6.4/7.0 based on this template, complete with Bootstrap, Doctrine, etc.  Alternatively, another demo that is based on this template but is more complete, something with a database and a 3rd-party js/css framework.

Thanks for releasing this.  Lots to learn, the Castor Task Runner looks pretty exciting.

**Comment by COil** [comment]
Hello, for `ux/live-components` , we must see if we will able to use them without using webpack. The goal here is really to avoid using yarn, npm...

**Comment by tacman** [comment]
AGREED!  I used this template as a base for creating a repo (https://github.com/tacman/micro-calendar) to report a bug (https://github.com/fullcalendar/fullcalendar/issues/7472), and have some comments about my experience with using the template.  

I'm happy to make this PRs if you agree with them.

* The default environment should be dev, like the default Symfony template
* debug-bundle and profiler-pack should be in the require-dev section of composer.  

This is what Symfony users are used to, I've deployed the template 4 times and it's always the first thing I do.

More significant, though, is that I think that the bulk of the demo should be in its own repo, with instructions to recreate.  The menu items should be minimal and common, like a link to the github repo and a link to the home page, and my a separate link to composer.json.  Or maybe simply comment out some of the things that aren't used.

The README could then contain instructions on how to use the template to quickly get up and running.  Something like

* create a project from the template
* Update brand and brand_html in services.yaml (though I'd prefer they be environment variables, in which case)
* change the brand and brand_html configuration in .env (or overwrite them in create .env.local)
* in base.html.twig, configure the menu items in the dropdown for your application
* also configure the footer

I'd like the navbar and footer to be configurable somewhere besides base.html.twig, though that starts to get complicated.  I've been playing around with the Twig UX components and how they can used with themes, so something like

For example, an anonymous component can be easily set up by creating 

```twig
{# templates/components/footer.html.twig #}
        <footer>
            <h5>{{ brand|raw }}</h5>

            <grid>
                <div fx="" col="3/4">
                    A Symfony minimalist application template by <a att="" href="https://strangebuzz.com" target="_blank">COil/Strangebuzz 🐝</a>
                    <br/>
                    Symfony ™ is a trademark of <a target="_blank" att="" href="https://symfony.com/license">Symfony SAS</a>.
                </div>

                <div fx="" col="1/4" txt="r">
                    <a att="" href="https://github.com/strangebuzz/MicroSymfony" target="_blank"><img alt="GitHub" class="github-icon" src="{{ asset('vendor/barecss/icons/github.svg') }}"></a>
                </div>
            </grid>
        </footer>
```

And then in base.html.twig, simply add 

```twig
<twig:footer />
```

Of course, the idea would be to make the footer customizable

```twig
<twig:footer :brand="brand" gitRepo="strangebuzz/MicroSymfony">
</twig:footer>

and change footer.html.twig to 

<a att="" href="https://github.com/{{ gitRepo }}" target="_blank"><img alt="GitHub" class="github-icon" src="{{ asset('vendor/barecss/icons/github.svg') }}"></a>

```

I'd like to use this template as the base for projects that I work on, almost all of which have bootstrap for styling and use the Symfony security-bundle for authentication.  And the menu system, as mentioned before.  And https://github.com/zenstruck/console-extra for creating easy commands, which will play very nicely with the task runner.

I'm still not resolved about how to use a github template vs. a theme bundle.  

I quite like this template, the task runner and selected bundles and such make it a great starting point.

And at some level, this is a warmup for Symfony 6.4 and 7, when assetmapper is no longer experimental and supports stylesheets.   I can't wait to drop yarn install && yarn dev from my installation instructions!
